freqtrade/data/btanalysis.py

```diff
@@ -213,3 +221,93 @@ def calculate_max_drawdown(trades: pd.DataFrame, *, date_col: str = 'close_time'
     low_date = profit_results.loc[max_drawdown_df['drawdown'].idxmin(), date_col]
 
     return abs(min(max_drawdown_df['drawdown'])), high_date, low_date
+
+
+def calculate_outstanding_balance(
+    results: pd.DataFrame,
+    timeframe: str,
+    min_date: datetime,
+    max_date: datetime,
+    hloc: Dict[str, pd.DataFrame],
+    slippage=0,
+) -> pd.DataFrame:
+    """
+    Sums the value of each trade (both open and closed) on each candle
+    :param results: Results Dataframe
+    :param timeframe: Frequency used for the backtest
+    :param min_date: date of the first trade opened (results.open_time.min())
+    :param max_date: date of the last trade closed (results.close_time.max())
+    :param hloc: historical DataFrame of each pair tested
+    :slippage: optional profit value to subtract per trade
+    :return: DataFrame of outstanding balance at each timeframe
+    """
+    timedelta = pd.Timedelta(timeframe)
+
+    date_index: pd.DatetimeIndex = pd.date_range(
+        start=min_date, end=max_date, freq=timeframe, normalize=True
+    )
+    balance_total = []
+    for pair in hloc:
+        pair_candles = hloc[pair].set_index("date").reindex(date_index)
+        # index becomes open_time
+        pair_trades = (
+            results.loc[results["pair"].values == pair]
+            .set_index("open_time")
+            .resample(timeframe)
+            .asfreq()
+            .reindex(date_index)
+        )
+        open_rate = pair_trades["open_rate"].fillna(0).values
+        open_time = pair_trades.index.values
+        close_time = pair_trades["close_time"].values
+        close = pair_candles["close"].values
+        profits = pair_trades["profit_percent"].values - slippage
```

Comment:
This column does not exist - neither in data loaded from a backtest file, nor from data loaded form database.

Reply:
mmmh? https://github.com/freqtrade/freqtrade/blob/25d9001f8864d6a641037f7d6f8e0177df18d752/freqtrade/optimize/optimize_reports.py#L21-L24

Reply:
nope - this is how it's internally called - but when loading data from file, it's aligned to the database.

https://github.com/freqtrade/freqtrade/blob/25d9001f8864d6a641037f7d6f8e0177df18d752/freqtrade/data/btanalysis.py#L19-L20